test-suite/daycounters.cpp

```diff
@@ -174,15 +205,189 @@ void DayCounterTest::testActualActual() {
+
+	Schedule schedule = MakeSchedule()
+		.from(fromDate)
+		.withFirstDate(firstCoupon)
+		.to(Date(28, February, 2026))
+		.withFrequency(Semiannual)
+		.withCalendar(calendar)
+		.withConvention(Unadjusted)
+		.backwards().endOfMonth(true);
+
+	Date testDate = schedule.date(1);
+	DayCounter dayCounter = ActualActual(ActualActual::ISMA, schedule);
+	DayCounter dayCounterNoSchedule = ActualActual(ActualActual::ISMA);
+	
+	Date referencePeriodStart = schedule.date(1);
+	Date referencePeriodEnd = schedule.date(2); 
+
+	//Test
+	QL_ASSERT(
+		dayCounter.yearFraction(referencePeriodStart, referencePeriodStart) == 0.0, "This should be zero."
+	);
+	QL_ASSERT(
+		dayCounterNoSchedule.yearFraction(referencePeriodStart, referencePeriodStart) == 0.0, "This should be zero"
+	);
+	QL_ASSERT(
+		dayCounterNoSchedule.yearFraction(referencePeriodStart, referencePeriodStart, referencePeriodStart, referencePeriodStart) == 0.0, "This should be zero"
+	);
+	QL_ASSERT(
+		dayCounter.yearFraction(referencePeriodStart, referencePeriodEnd) == 0.5, "This should be exact using schedule; " << referencePeriodStart << " to " << referencePeriodEnd << "Should be 0.5"
+	);
+	QL_ASSERT(
+		dayCounterNoSchedule.yearFraction(referencePeriodStart, referencePeriodEnd) == dayCounterNoSchedule.dayCount(referencePeriodStart, referencePeriodEnd)/365.0, 
+		"This should be exact with no schedule; " << referencePeriodStart << " to " << referencePeriodEnd << ". This should be " << dayCounterNoSchedule.dayCount(referencePeriodStart, referencePeriodEnd)/365.0 << " /365 but was actually : " << dayCounterNoSchedule.yearFraction(referencePeriodStart, referencePeriodEnd)
+	);
+	QL_ASSERT(
+		dayCounterNoSchedule.yearFraction(referencePeriodStart, referencePeriodEnd, referencePeriodStart, referencePeriodEnd) == 0.5, "This should be exact for explicit reference periods with no schedule";
+	);
+	
+
+	while (testDate < referencePeriodEnd) {
+		double difference = dayCounter.yearFraction(testDate, referencePeriodEnd, referencePeriodStart, referencePeriodEnd) -
+			dayCounter.yearFraction(testDate, referencePeriodEnd);
+		if (std::fabs(difference) > 1.0e-10) {
+			BOOST_ERROR("FAILED TO CORRECTLY USE THE SCHEDULE TO FIND THE REFERENCE PERIOD FOR ACT ACT");
+		};
+		testDate = calendar.advance(testDate, 1, Days);
+	}
+
+	//Test long first coupon
+	Real caclulatedYearFraction = dayCounter.yearFraction(fromDate, firstCoupon);
+	Real expectedYearFraction = 0.5 + ((Real) dayCounter.dayCount(fromDate, quasiCoupon))/(2*dayCounter.dayCount(quasiCoupon2, quasiCoupon));
+	Real secondCalculation = dayCounterNoSchedule.yearFraction(fromDate, quasiCoupon, quasiCoupon2, quasiCoupon);
+
+	QL_ASSERT(
+		std::fabs(caclulatedYearFraction - expectedYearFraction) < 1.0e-10, 
+		"These Year Fractions Should be Identical: expected " << expectedYearFraction << " but calculated " << caclulatedYearFraction
+	);
+	QL_ASSERT(
+		std::fabs(caclulatedYearFraction - expectedYearFraction) < 1.0e-10, 
```

Comment:
Same check as above.  I guess you wanted to compare `secondCalculation` instead?

Reply:
(If so, the test doesn't pass.)